### PR TITLE
feat: release overview content

### PR DESCRIPTION
**Pull request by CzarekDryl**
## Description

- Opens the Release payment modal with the users possible decision methods
- If funded, the owner can release the funds for the payment, starting the claim delay.
- If any claim delays are 0, then those payments will happen immediately on release and also release the creators stake if present.
- Alternatively, users can use available decision methods, including "Admin" tiered permissions or higher (Arbitration on the contracts)

### CODE REVIEW

**Comment by AdrianDudko** [comment]
Waiting for https://github.com/JoinColony/colonyCDapp/issues/2086

**Comment by jakubcolony** [comment]
@AdrianDudko #2086 has just been merged. On a side note, once a PR is up, the work dependent on it can be started as long as it's checked out from that branch, without waiting for a merge.

File: src/components/v5/common/CompletedAction/partials/PaymentBuilder/partials/PaymentBuilderWidget/utils.ts [hunk 1]
```diff
@@ -9,7 +12,47 @@ export const getExpenditureStep = (status?: ExpenditureStatus) => {
...
+          amounts[payout.tokenAddress] = BigNumber.from(
+            amounts[payout.tokenAddress] ?? 0,
+          ).add(payout.amount);
+        }
+      });
+
+      return amounts;
+    },
+    {},
+  );
```
Comment (rdig) [inline src/components/v5/common/CompletedAction/partials/PaymentBuilder/partials/PaymentBuilderWidget/utils.ts:49]: This is improper use of `reduce`. There no case where you actually modify or increment the accumulator, you always return the the same object.

This is at best a use case for `map` rather.

File: src/components/v5/common/CompletedAction/partials/PaymentBuilder/partials/PaymentBuilderWidget/utils.ts [hunk 2]
```diff
@@ -9,7 +12,47 @@ export const getExpenditureStep = (status?: ExpenditureStatus) => {
...
+  if (!expenditure.balances) {
+    return false;
+  }
+
+  const amountsByToken = expenditure.slots.reduce<Record<string, BigNumber>>(
+    (amounts, slot) => {
+      slot.payouts?.forEach((payout) => {
+        if (!payout.isClaimed) {
+          // eslint-disable-next-line no-param-reassign
+          amounts[payout.tokenAddress] = BigNumber.from(
```
Comment (rdig) [inline src/components/v5/common/CompletedAction/partials/PaymentBuilder/partials/PaymentBuilderWidget/utils.ts:40]: Please be mindful of our eslint rules

In this case a better approach would have been to just declare a new variable which would hold the value you need

**Review by rdig — CHANGES_REQUESTED** [review]
Mostly good, however the code inside `isExpenditureFullyFunded` needs to be refactored some.

Screenshot from 2024-03-30 18-45-31

Also, I am guessing this PR doesn't actually bring in the content, just creates a placeholder for it _(based on what I'm seeing inside the Release step)

**Comment by CzarekDryl** [comment]
@rdig I've fixed content inside release overview box. Please check it once again.

File: src/components/v5/common/CompletedAction/partials/PaymentBuilder/partials/PaymentBuilderWidget/utils.ts [hunk 3]
```diff
@@ -9,7 +12,48 @@ export const getExpenditureStep = (status?: ExpenditureStatus) => {
...
+export const isExpenditureFullyFunded = (expenditure?: Expenditure | null) => {
+  if (!expenditure) {
+    return false;
+  }
+
+  if (!expenditure.balances) {
+    return false;
+  }
+
+  const amountsByToken = expenditure.slots.map((slot) => {
```
Comment (bassgeta) [inline src/components/v5/common/CompletedAction/partials/PaymentBuilder/partials/PaymentBuilderWidget/utils.ts:35]: nitpicky, but could we name this something like `slotAmountsByToken`? `amountsByToken` sounds like an object with token addresses as keys and amounts as values tbh :D

File: src/components/v5/common/CompletedAction/partials/PaymentBuilder/partials/PaymentBuilderWidget/utils.ts [hunk 4]
```diff
@@ -9,7 +12,48 @@ export const getExpenditureStep = (status?: ExpenditureStatus) => {
...
+          amounts[payout.tokenAddress] ?? 0,
+        ).add(payout.amount);
+      }
+    });
+
+    return amounts;
+  });
+
+  return amountsByToken.every((token) => {
+    const [tokenAddress] = Object.keys(token);
```
Comment (bassgeta) [inline src/components/v5/common/CompletedAction/partials/PaymentBuilder/partials/PaymentBuilderWidget/utils.ts:50]: I am not sure I am a fan of this, could we construct the object differently so we don't have to take the first key and then checking the amount?

**Review by bassgeta — CHANGES_REQUESTED** [review]
Nice start on this! I would like it if we somehow simplify the slot amount checking logic so it's more readable (if possible) :)

**Review by bassgeta — APPROVED** [review]
Reads much nicer now, thanks for the quick update!

**Review by rdig — APPROVED** [review]
Good from my side as well. Nice work!

**Review by jakubcolony — APPROVED** [review]
Looks good @CzarekDryl 👍 

image

**Pull request merged**